frontend/app/src/features/auth/api/useGetGithubAuthUrl.ts

```diff
@@ -0,0 +1,29 @@
+import { backendApi } from "@/lib/axios";
+import { makeSuccessResponseSchema } from "@/lib/api";
+import { useQuery } from "react-query";
+import { z } from "zod";
+
+const GET_GITHUB_AUTH_URL_KEY = "github-auth-url";
+
+const getGithubAuthUrlResponseSchema = makeSuccessResponseSchema(
+  z.object({
+    url: z.string().url(),
+  }),
+);
+
+const getGithubAuthUrl = async () => {
+  const { data } = await backendApi.get("/auth/github/authorize");
```

Comment:
Can make constants for these also

Reply:
completed in [5a14905](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/pull/8/commits/5a14905ed45bc88b6dc264136d26f455440d414e)